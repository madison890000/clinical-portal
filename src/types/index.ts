export type IPerson = {
    title?: string;
    firstName: string;
    preferredName?: string;
    middleName?: string;
    familyName: string;
    suffix?: string;
}

export type IClinician = {
    username: string;
    role: string;
} & IPerson;

export type IPatient = {
    age: number;
    sex: 'Male' | 'Female' | 'Unknown' | 'Indeterminate';
} & IPerson;

export type PatientSummary = {
    id: string;
    name: string;
}

export type IPatientList = {
    patients: PatientSummary[];
}

export type IBaseErrorResponse = {
    httpStatusCode: number;
    errorMessage: string;
}

export enum LoginStatus {
    NotLogin,
    Login,
    TokenExpired
}
