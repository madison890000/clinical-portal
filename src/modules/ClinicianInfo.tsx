import Name from '../components/Name';
import { IClinician } from '../types';
import Logout from './Logout';
import { Card } from '@mui/material';
import InfoItem from '../components/InfoItem';
import styles from './ClinicianInfo.module.scss';

const ClinicianInfo = ({ ...clinician }: IClinician) => {
    return (
        <Card className={styles.container}>
            <div className={styles.name}>
                <Name {...clinician} />
                <Logout />
            </div>
            <InfoItem name="username" value={clinician?.username} />
            <InfoItem name="role" value={clinician?.role} />
        </Card>
    );
};

export default ClinicianInfo;
