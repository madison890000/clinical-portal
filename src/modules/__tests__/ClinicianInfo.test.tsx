import React from 'react';
import ClinicianInfo from '../ClinicianInfo';
import renderer from 'react-test-renderer';
import { IClinician } from '../../types';

jest.mock('../Logout.tsx');
test('renders ClinicianInfo correctly', () => {
    const demoClinician = {
        username: 'joshs',
        role: 'General Practitioner',
        title: 'Dr',
        firstName: 'Joshua',
        preferredName: 'Josh',
        familyName: 'Smith'
    } as IClinician;
    const ClinicianInfoDom = renderer.create(<ClinicianInfo {...demoClinician} />).toJSON();
    expect(ClinicianInfoDom).toMatchSnapshot();
});
