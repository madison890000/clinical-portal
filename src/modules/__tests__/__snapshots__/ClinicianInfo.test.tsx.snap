// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ClinicianInfo correctly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root container css-bhp9pd-MuiPaper-root-MuiCard-root"
>
  <div
    className="name"
  >
    <div
      className="container"
    >
      <div
        className="text"
      >
        Dr
      </div>
      <div
        className="text"
      >
        (Josh)
      </div>
      <div
        className="text"
      />
      <div
        className="text"
      >
        Smith
      </div>
      <div
        className="text"
      />
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Logout
    </button>
  </div>
  <div
    className="container"
  >
    <div
      className="name"
    >
      Username
      :
    </div>
    <span
      className="MuiTypography-root MuiTypography-overline css-1kdyykn-MuiTypography-root"
    >
      joshs
    </span>
  </div>
  <div
    className="container"
  >
    <div
      className="name"
    >
      Role
      :
    </div>
    <span
      className="MuiTypography-root MuiTypography-overline css-1kdyykn-MuiTypography-root"
    >
      General Practitioner
    </span>
  </div>
</div>
`;
